eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/widgets/ClearableEditText.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.widgets;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.os.Build;
+import android.text.Editable;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.EditText;
+
+import com.schedjoules.eventdiscovery.R;
+
+
+/**
+ * {@link EditText} that has an 'X' right side compound drawable as a clear button.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class ClearableEditText extends EditText
+{
+    public ClearableEditText(Context context)
+    {
+        super(context);
+        init();
+    }
+
+
+    public ClearableEditText(Context context, AttributeSet attrs)
+    {
+        super(context, attrs);
+        init();
+    }
+
+
+    public ClearableEditText(Context context, AttributeSet attrs, int defStyleAttr)
+    {
+        super(context, attrs, defStyleAttr);
+        init();
+    }
+
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public ClearableEditText(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes)
+    {
+        super(context, attrs, defStyleAttr, defStyleRes);
+        init();
+    }
+
+
+    private void init()
+    {
+        addTextChangedListener(new AbstractTextWatcher()
+        {
+            @Override
+            public void afterTextChanged(Editable text)
+            {
+                int clearIcon = text.length() > 0 ? R.drawable.schedjoules_ic_clear : 0;
+                setCompoundDrawablesWithIntrinsicBounds(0, 0, clearIcon, 0);
+            }
+        });
+
+        // http://stackoverflow.com/a/29214019/4247460
+        setOnTouchListener(new View.OnTouchListener()
+        {
+            @Override
+            public boolean onTouch(View v, MotionEvent event)
+            {
+                if (event.getAction() == MotionEvent.ACTION_UP)
```

Comment:
IMO this approach is a bit too simple, because it will wipe the field even if you don't tap on the "X" but swipe onto it before lifting your finger. You should at least test if the finger hit the area in the previous `ACTION_DOWN` event.